### ISSUE TITLE
poor performance on large files

### ISSUE THREAD

**Issue by drvid** [mrhanlon/less-than-slash#48]
Hi it just so happens I am using this on a very big HTML file (almost 10,000 lines).
When I type ```</``` towards the bottom of the document, it can take up to 10 seconds to complete closing the tag. Sometimes the whole editor goes unresponsive even.

**Comment by MarcoThePoro** [comment]
Yeah there's a lot of regex going on. Maybe we could investigate parsing backwards from the insertion point instead of right from the start? Idk how to do perf testing...

**Comment by MarcoThePoro** [comment]
Maybe we could cache the stack of tags so we don't have to parse the entire document on every completion, or find a way to partially parse the document in the background in anticipation.

We could also investigate using a more performant parser than regex, but it's harder to add custom syntaxes eg. jsx mustache

**Comment by kirschkern** [comment]
I have the same issue. Using a javascript file with 2000 lines, typing gets really slow and the editor often freezes for a couple of seconds. I've disabled the plugin right now, as editing such files is a real pain. Hope to see this fixed as I like the plugin and use it all the time.
